src/openforms/submissions/tests/form_logic/test_modify_variables.py

```diff
@@ -94,9 +94,8 @@ def test_modify_variable_related_to_step_being_edited(self):
 
         evaluate_form_logic(submission, submission_step2)
 
-        variables_state = submission.load_submission_value_variables_state()
-        variable = variables_state.variables["nTotalBoxes"]
-        self.assertEqual(7, variable.value)
+        state = submission.load_submission_value_variables_state()
+        self.assertEqual(state.get_data(include_unsaved=True)["nTotalBoxes"], 7)
```

Comment:
Why is this change needed? I thought that `variables_state.variables` includes unsaved vars, so the test should still work without the change, right?